js/tab.js

```diff
@@ -171,6 +180,32 @@ define([
     return syntaxValue;
   }
   
+  Tab.prototype.getWorker = function() {
+    return new Promise((resolve, reject) => {
+      // if worker is off, return null
+      if (!Settings.get("user").useWorker) return resolve(null);
+      
+      var checkLimit = 20;
+      var findWorker = () => {
+        // if worker was turned while we were waiting on it, return null
+        if (!Settings.get("user").useWorker) return resolve(null);
+      
+        // return the worker if we found it
+        if (this.$worker) {
+          return resolve(this.$worker);
+        }
+        
+        // see if we've checked too many times
+        if (checkLimit-- < 0) return reject(new Error('Could not find worker.'));
+        
+        // otherwise, go to sleep and then check again
+        setTimeout(findWorker, 100);
```

Comment:
Yeah, I was thinking that too. Under https://github.com/thomaswilburn/Caret/pull/526/files#diff-efb69290d99ac658725bf3c4691f9851R144 I originally just checked if workers were in use and if so tried to message it. However settings were not getting set on startup, though issuing an `init:reset` a few seconds after would trigger the settings updating.

```javascipt
this.setUseWorker(userConfig.useWorker);

if (userConfig.useWorker) {
  // .$worker would sometimes be null here
  this.$worker.send(...);
}
```